app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,48 @@
+class OrdersController < ApplicationController
+  before_action :find_order, only: [:cart, :checkout, :update, :receipt]
+  before_action :check_access, only: [:cart, :checkout, :receipt]
+
+
+  def cart; end
+
+  def checkout; end # of a particular item in a cart; increases the quantity of an item in the cart
+
+  def update
+    # add buyer info to order & change status
+    # handling for error messages / bad input if/else type thing
+    @order.update(checkout_params)
+
+    redirect_to receipt_path
+  end
+
+  def receipt
+    if @order.status == "paid"
+      render :receipt
+
+      # will this work? no?
+      reset_session
```

Comment:
Yeah, I'm pretty sure it will. If a user is logged in, though, this would immediately log them out. Is that intended behavior?